cpmpy/solvers/z3.py

```diff
@@ -379,7 +379,12 @@ def _z3_expr(self, cpm_con):
                 elif cpm_con.name == "mul":
                     return x * y
                 elif cpm_con.name == "div":
-                    return x / y
+                    # z3 rounds towards negative infinity, need this hack when result is negative
+                    return z3.If(z3.And(x >= 0, y >= 0), x / y,
+                           z3.If(z3.And(x <= 0, y <= 0), -x / -y,
+                           z3.If(z3.And(x >= 0, y <= 0), -(x / -y),
+                           z3.If(z3.And(x <= 0, y >= 0), -(-x / y), 0))))
```

Comment:
ah damn that's nice!
Do we ever reach that 0 case?

Reply:
Normally no, as we have safened away the 0-s in the denominator